Cubical/HITs/SmashProduct/Base.agda

```diff
@@ -3,23 +3,45 @@ module Cubical.HITs.SmashProduct.Base where
 
 open import Cubical.Foundations.Prelude
 open import Cubical.Foundations.Pointed
+open import Cubical.Foundations.Isomorphism
+open import Cubical.HITs.Pushout.Base
+open import Cubical.Data.Unit
+open import Cubical.Data.Prod
+open import Cubical.HITs.Wedge
+open import Cubical.Foundations.Function
+open import Cubical.Foundations.GroupoidLaws
 
-data Smash (A B : Pointed₀) : Type₀ where
+data Smash {ℓ ℓ'} (A : Pointed ℓ) (B : Pointed ℓ') : Type (ℓ-max ℓ ℓ') where
   basel : Smash A B
   baser : Smash A B
   proj  : (x : typ A) → (y : typ B) → Smash A B
   gluel : (a : typ A) → proj a (pt B) ≡ basel
   gluer : (b : typ B) → proj (pt A) b ≡ baser
 
+private
+  variable
+    ℓ ℓ' ℓ'' ℓ''' : Level
+    A : Pointed ℓ
+    B : Pointed ℓ'
+    C : Pointed ℓ''
+    D : Pointed ℓ'''
```

Comment:
@Saizan : I'm always confused about how `variable` works in Agda, but I don't think you need all of these levels to be different?

Reply:
I can't remember, did we come to a conclusion here? Should I leave it as it is or not?

Reply:
Yeah, you can use \ell for A-D and get the same effect. The only time when you need to have different levels is in statements where you want more than one. So please reduce the number of levels to the actual number that you need in the file.